ewatercycle/models/lisflood.py

```diff
@@ -0,0 +1,398 @@
+                self.end = datetime.utcfromtimestamp(dataset.coords['time'][-1].values.astype('O') / 1e9)
+        elif isinstance(forcing, ForcingData):
+            # key is cmor var name and value is path to NetCDF file
+            self.forcing_files = dict()
+            data_files = list(forcing.recipe_output.values())[0].data_files
+            for data_file in data_files:
+                dataset = data_file.load_xarray()
+                var_name = list(dataset.data_vars.keys())[0]
+                self.forcing_files[var_name] = data_file.filename.name
+                self.forcing_dir = data_file.filename.parent
+                # get start and end date of time dimension
+                # TODO converting numpy.datetime64 to datetime object is ugly, find better way
+                self.start = datetime.utcfromtimestamp(dataset.coords['time'][0].values.astype('O') / 1e9)
+                self.end = datetime.utcfromtimestamp(dataset.coords['time'][-1].values.astype('O') / 1e9)
+        else:
+            raise TypeError(
+                f"Unknown forcing type: {forcing}. Please supply either a Path or ForcingData object."
+            )
+
+    def _create_lisflood_config(self) -> str:
+        """Create lisflood config file"""
+        cfg = XmlConfig(self.parameterset.config_template)
+
+        settings = {
+            "CalendarDayStart": self.start.strftime("%d/%m/%Y 00:00"),
+            "StepStart": "1",
+            "StepEnd": str((self.end - self.start).days),
+            "PathRoot": f"{self.parameterset.root}",
+            "MaskMap": f"{self.parameterset.mask}".rstrip('.nc'),
+            "PathMeteo": f"{self.forcing_dir}",
+            "PathOut": f"{self.work_dir}",
+        }
+
+        timestamp = f"{self.start.year}_{self.end.year}"
+
+        for textvar in cfg.config.iter("textvar"):
+            textvar_name = textvar.attrib["name"]
+
+            # general settings
+            for key, value in settings.items():
+                if key in textvar_name:
+                    textvar.set("value", value)
+
+            # input for lisflood
+            if "PrefixPrecipitation" in textvar_name:
+                textvar.set("value", self.forcing_files['pr'].rstrip('.nc'))
+            if "PrefixTavg" in textvar_name:
+                textvar.set("value", self.forcing_files['tas'].rstrip('.nc'))
+
+            # output of lisvap
+            for map_var, prefix in MAPS_PREFIXES.items():
+                if prefix['name'] in textvar_name:
+                    textvar.set(
+                        "value",
+                        f"lisflood_{prefix['value']}_{timestamp}",
+                    )
+                if map_var in textvar_name:
+                    textvar.set('value', f"$(PathOut)/$({prefix['name']})")
+
+        # Write to new setting file
+        lisflood_file = f"{self.work_dir}/lisflood_setting_{timestamp}.xml"
```

Comment:
Why do we need a timestamp in the settings filename?

Reply:
Good point, it is removed.